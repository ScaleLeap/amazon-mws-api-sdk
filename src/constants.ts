/**
 * The version value is automatically replaced during builds.
 */
export const USER_AGENT = '@scaleleap/amazon-mws-api-sdk/1.9.0'
