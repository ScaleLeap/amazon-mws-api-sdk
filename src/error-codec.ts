import { Codec, exactly, oneOf, optional, string } from 'purify-ts/Codec'

export enum CodeEnum {
  InputStreamDisconnected,
  InvalidParameterValue,
  AccessDenied,
  InvalidAccessKeyId,
  SignatureDoesNotMatch,
  InvalidAddress,
  InternalError,
  'Internal Error',
  QuotaExceeded,
  RequestThrottled,
  ResourceNotFound,
  ScheduledPackageAlreadyExists,
  RegionNotSupported,
  ScheduleWindowExpired,
  InvalidOrderState,
  PickupSlotNotAvailable,
  AccessToFeedProcessingResultDenied,
  ContentMD5Missing,
  ContentMD5DoesNotMatch,
  FeedCanceled,
  FeedProcessingResultNoLongerAvailable,
  FeedProcessingResultNotReady,
  InputDataError,
  InvalidFeedSubmissionId,
  InvalidFeedType,
  InvalidRequest,
  NonRetriableInternalError,
  RetriableInternalError,
  AccessToReportDenied,
  InvalidReportId,
  InvalidReportRequestId,
  InvalidReportType,
  InvalidScheduleFrequency,
  ReportNoLongerAvailable,
  ReportNotReady,
  DependencyFatalException,
  DependencyRetriableException,
  DependencyUnauthorizedException,
  InternalErrorFatalException,
  InvalidInputFatalException,
}

export const Error = Codec.interface({
  Type: string,
  Code: oneOf(Object.keys(CodeEnum).map((x) => exactly(x)) as [Codec<string>, ...Codec<string>[]]),
  Message: string,
  Detail: optional(string),
})

export const MWSApiError = Codec.interface({
  ErrorResponse: Codec.interface({
    Error,
    RequestID: optional(string),
    RequestId: optional(string),
  }),
})
