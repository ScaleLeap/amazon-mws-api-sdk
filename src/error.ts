/* eslint-disable max-classes-per-file */
export class MWSError extends Error {
  constructor(...parameters: string[]) {
    // Propagate some vendor-specific arguments
    super(...parameters)

    // If someone downlevels the compilation target to es5
    Object.setPrototypeOf(this, MWSError.prototype)

    // Maintains proper stack trace (only available on V8)
    if (Error.captureStackTrace) {
      Error.captureStackTrace(this, MWSError)
    }
  }
}

export class HttpError extends MWSError {
  public message = 'Encountered an error while sending a request'

  constructor(public error: unknown, ...parameters: string[]) {
    super(...parameters)
    Object.setPrototypeOf(this, HttpError.prototype)

    if (Error.captureStackTrace) {
      Error.captureStackTrace(this, HttpError)
    }
  }
}

export class ParsingError extends MWSError {
  public message = 'Encountered an error while parsing a response'

  constructor(public error: string, ...parameters: string[]) {
    super(...parameters)
    Object.setPrototypeOf(this, ParsingError.prototype)

    if (Error.captureStackTrace) {
      Error.captureStackTrace(this, ParsingError)
    }
  }
}
/* eslint-enable max-classes-per-file */
