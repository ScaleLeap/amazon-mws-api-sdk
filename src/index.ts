export * from './http'
export * from './mws'
export * from './sections/sellers'
export * from './sections/orders'
export * from './sections/products/products'
export * from './sections/fulfillment-inventory'
export * from './sections/reports'
export * from './error'
export * from '@scaleleap/amazon-marketplaces'
