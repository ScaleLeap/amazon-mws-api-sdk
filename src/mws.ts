import { HttpClient } from './http'
import { FulfillmentInventory } from './sections/fulfillment-inventory'
import { Orders } from './sections/orders'
import { Products } from './sections/products/products'
import { Reports } from './sections/reports'
import { Sellers } from './sections/sellers'

export class MWS {
  private _sellers!: Sellers

  private _orders!: Orders

  private _fulfillmentInventory!: FulfillmentInventory

  private _products!: Products

  private _reports!: Reports

  constructor(private httpClient: HttpClient) {}

  get sellers() {
    if (!this._sellers) {
      this._sellers = new Sellers(this.httpClient)
    }

    return this._sellers
  }

  get orders() {
    if (!this._orders) {
      this._orders = new Orders(this.httpClient)
    }

    return this._orders
  }

  get fulfillmentInventory() {
    if (!this._fulfillmentInventory) {
      this._fulfillmentInventory = new FulfillmentInventory(this.httpClient)
    }

    return this._fulfillmentInventory
  }

  get products() {
    if (!this._products) {
      this._products = new Products(this.httpClient)
    }

    return this._products
  }

  get reports() {
    if (!this._reports) {
      this._reports = new Reports(this.httpClient)
    }

    return this._reports
  }
}
