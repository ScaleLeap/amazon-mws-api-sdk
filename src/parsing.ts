/** A collection of parsing utils */

export const ensureArray = <T>(x: T | T[]): T[] => (Array.isArray(x) ? x : [x])

export const parseBoolean = (x: 'Yes' | 'No'): boolean => {
  switch (x) {
    case 'Yes':
      return true
    case 'No':
      return false
    default:
      throw new Error('TODO, but should be library-specific error')
  }
}
