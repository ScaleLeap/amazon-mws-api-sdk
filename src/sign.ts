import crypto from 'crypto'

export const sign = (queryString: string, secret: string): string => {
  const hmac = crypto.createHmac('sha256', secret)
  hmac.update(queryString, 'utf8')
  return hmac.digest('base64')
}
