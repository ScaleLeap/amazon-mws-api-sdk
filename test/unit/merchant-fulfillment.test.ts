import { ParsingError } from '../../src'
import {
  GetEligibleShippingServicesParameters,
  ShippingServiceOptions as ShippingServiceOptionsInterface,
  Weight as WeightInterface,
} from '../../src/sections/merchant-fulfillment/type'
import { createMockHttpClient, mockMwsFail, mockMwsServiceStatus, parsingError } from '../utils'

const mockAddress = {
  Name: '',
  AddressLine1: '',
  Email: '',
  City: '',
  PostalCode: '',
  CountryCode: '',
  Phone: '',
}

describe('merchant-fulfillment', () => {
  describe('getAdditionalSellerInputs', () => {
    const parameters = {
      OrderId: '',
      ShippingServiceId: '',
      ShipFromAddress: mockAddress,
    }

    it('returns shipment level fields and item level fields if succesful', async () => {
      expect.assertions(1)

      const mockGetAdditionalSellerInputs = createMockHttpClient(
        'merchant_get_additional_seller_inputs',
      )

      expect(
        await mockGetAdditionalSellerInputs.merchantFulfillment.getAddtionalSellerInputs(
          parameters,
        ),
      ).toMatchSnapshot()
    })

    it("throws a parsing error when the status response isn't valid", async () => {
      expect.assertions(1)

      await expect(() =>
        mockMwsFail.merchantFulfillment.getAddtionalSellerInputs(parameters),
      ).rejects.toStrictEqual(new ParsingError(parsingError))
    })
  })

  describe('getEligiblShippingServices', () => {
    const PackageDimensions = {
      PredefinePackageDimensions: 'FedEx_Box_10kg',
    }

    const Weight: WeightInterface = {
      Value: 1,
      Unit: 'ounces',
    }

    const ShippingServiceOptions: ShippingServiceOptionsInterface = {
      DeliveryExperience: 'DeliveryConfirmationWithAdultSignature',
      CarrierWillPickUp: false,
    }

    const ShipmentRequestDetails = {
      AmazonOrderId: '',
      SellerOrderId: '',
      ItemList: [],
      ShipFromAddress: mockAddress,
      PackageDimensions,
      Weight,
      MustArriveByDate: new Date(),
      ShipDate: new Date(),
      ShippingServiceOptions,
    }

    const parameters: GetEligibleShippingServicesParameters = {
      ShipmentRequestDetails,
    }

    it('returns lists of shipping services if succesful with test from C# mocks', async () => {
      expect.assertions(1)

      const mockGetEligibleShippingServices = createMockHttpClient(
        'merchant_fulfillment_get_eligible_shipping_services_no_from_c_sharp',
      )

      expect(
        await mockGetEligibleShippingServices.merchantFulfillment.getEligibleShippingServices(
          parameters,
        ),
      ).toMatchSnapshot()
    })

    it('returns lists of shipping services if succesful with no additional input', async () => {
      expect.assertions(1)

      const mockGetEligibleShippingServices = createMockHttpClient(
        'merchant_fulfillment_get_eligible_shipping_services_no_additional_input',
      )

      expect(
        await mockGetEligibleShippingServices.merchantFulfillment.getEligibleShippingServices(
          parameters,
        ),
      ).toMatchSnapshot()
    })

    it('returns lists of shipping services if succesful with additional input', async () => {
      expect.assertions(1)

      const mockGetEligibleShippingServices = createMockHttpClient(
        'merchant_fulfillment_get_eligible_shipping_services_additional_input',
      )

      expect(
        await mockGetEligibleShippingServices.merchantFulfillment.getEligibleShippingServices(
          parameters,
        ),
      ).toMatchSnapshot()
    })

    it('throws a parsing error when the status response is not valid', async () => {
      expect.assertions(1)

      await expect(() =>
        mockMwsFail.merchantFulfillment.getEligibleShippingServices(parameters),
      ).rejects.toStrictEqual(new ParsingError(parsingError))
    })
  })

  describe('getServiceStatus', () => {
    it('returns a parsed model when the status response is valid', async () => {
      expect.assertions(1)

      expect(await mockMwsServiceStatus.merchantFulfillment.getServiceStatus()).toMatchSnapshot()
    })

    it('throws a parsing error when the status response is not valid', async () => {
      expect.assertions(1)

      await expect(() => mockMwsFail.merchantFulfillment.getServiceStatus()).rejects.toStrictEqual(
        new ParsingError(parsingError),
      )
    })
  })
})
