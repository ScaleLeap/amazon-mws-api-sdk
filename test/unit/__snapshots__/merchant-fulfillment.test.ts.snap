// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`merchant-fulfillment createShipment returns details of the shipment if succesful 1`] = `
Array [
  Object {
    "Shipment": Object {
      "AmazonOrderId": "903-1713775-3598252",
      "CreatedDate": 2015-09-23T20:11:12.908Z,
      "Insurance": Object {
        "CurrencyCode": "USD",
      },
      "ItemList": Array [
        Object {
          "ItemDescription": "Poster",
          "ItemWeight": Object {
            "Unit": "gram",
            "Value": 100,
          },
          "OrderItemId": "40525960574974",
          "Quantity": 1,
        },
      ],
      "Label": Object {
        "CustomTextForLabel": "ABC123",
        "Dimensions": Object {
          "Length": 11,
          "Unit": "inches",
          "Width": 8.5,
        },
        "FileContents": Object {
          "Checksum": "DmsWbJpdMPALN3jV4wHOrg==",
          "Contents": "H4sIAAAAAAAAAK16WbeqyrLmO2Pc/zBVRLG...
                        ...kI0Ycv76w378UfL6O6ev7UsAwmkWBfL6nycD/Bd+Zx3S8LwAA",
          "FileType": "application/zpl",
        },
        "LabelFormat": "ZPL203",
        "StandardIdForLabel": "AmazonOrderId",
      },
      "PackageDimensions": Object {
        "Height": 5,
        "Length": 5,
        "Unit": "inches",
        "Width": 5,
      },
      "ShipFromAddress": Object {
        "AddressLine1": "1234 Westlake Ave",
        "City": "Seattle",
        "CountryCode": "US",
        "Email": "String",
        "Name": "John Doe",
        "Phone": "2061234567",
        "PostalCode": "98121",
        "StateOrProvinceCode": "WA",
      },
      "ShipToAddress": Object {
        "AddressLine1": "321 Main St",
        "City": "Seattle",
        "CountryCode": "US",
        "Email": "",
        "Name": "Jane Smith",
        "Phone": "",
        "PostalCode": "98121-2778",
        "StateOrProvinceCode": "WA",
      },
      "ShipmentId": "6f77095e-9f75-47eb-aaab-a42d5428fa1a",
      "ShippingService": Object {
        "CarrierName": "FEDEX",
        "EarliestEstimatedDeliveryDate": 2015-09-24T10:30:00.000Z,
        "LatestEstimatedDeliveryDate": 2015-09-24T10:30:00.000Z,
        "Rate": Object {
          "CurrencyCode": "USD",
        },
        "ShipDate": 2015-09-23T20:10:56.829Z,
        "ShippingServiceId": "FEDEX_PTP_PRIORITY_OVERNIGHT",
        "ShippingServiceName": "FedEx Priority OvernightÂ®",
        "ShippingServiceOfferId": "HDDUKqtQVFetpBZAqx5c1yaCZ9vuFfND0kudyw3lLWCa
                    3mN2+zUOsRCAZS2oYt0ey6fXKdOAucmYVXR9LAkU9O9eys9V3FJCgvQ+vAqm4bSAzHVH04vD8oLy oUPKqxa8Otbek97Z85LFUSsfcUf8frNRMEfNLP8t5RyLXa4ZOjc=",
        "ShippingServiceOptions": Object {
          "CarrierWillPickUp": false,
          "DeclaredValue": Object {
            "CurrencyCode": "USD",
          },
          "DeliveryExperience": "DELIVERY_CONFIRMATION",
        },
      },
      "Status": "Purchased",
      "TrackingId": "794657111237",
      "Weight": Object {
        "Unit": "oz",
        "Value": 10,
      },
    },
  },
  Object {
    "quotaMax": 1000,
    "quotaRemaining": 999,
    "quotaResetOn": 2020-04-06T10:22:23.582Z,
    "requestId": "0",
    "timestamp": 2020-05-06T09:22:23.582Z,
  },
]
`;

exports[`merchant-fulfillment getAdditionalSellerInputs returns shipment level fields and item level fields if succesful 1`] = `
Array [
  Object {
    "ItemLevelFieldsList": Array [
      Object {
        "AdditionalInputs": Array [
          Object {
            "AdditionalInputFieldName": "ITEM_WEIGHT",
            "SellerInputDefinition": Object {
              "Constraints": Array [
                Object {
                  "ValidationRegEx": "^[0-9]{1,3}(.[0-9]{1,2})?$",
                  "ValidationString": "Please enter item weight (max 6
                                        digits)",
                },
              ],
              "DataType": "Weight",
              "InputDisplayText": "Item weight",
              "InputTarget": "ITEM_LEVEL",
              "IsRequired": true,
              "RestrictedSetValues": Array [],
              "StoredValue": Object {
                "DataType": "WEIGHT",
                "ValueAsBoolean": false,
                "ValueAsInteger": 0,
                "ValueAsWeight": Object {
                  "Unit": "G",
                  "Value": 600,
                },
              },
            },
          },
          Object {
            "AdditionalInputFieldName": "COUNTRY_OF_ORIGIN",
            "SellerInputDefinition": Object {
              "Constraints": Array [
                Object {
                  "ValidationRegEx": "^[a-zA-Z]{2}$",
                  "ValidationString": "Please enter country code of the country of
                                        manufacture (e.g. CN)",
                },
              ],
              "DataType": "String",
              "InputDisplayText": "Country of Origin",
              "InputTarget": "ITEM_LEVEL",
              "IsRequired": true,
              "RestrictedSetValues": Array [],
              "StoredValue": Object {
                "DataType": "STRING",
                "ValueAsBoolean": false,
                "ValueAsInteger": 0,
                "ValueAsString": "CN",
              },
            },
          },
          Object {
            "AdditionalInputFieldName": "ITEM_DESCRIPTION_TRANSLATED",
            "SellerInputDefinition": Object {
              "Constraints": Array [
                Object {
                  "ValidationRegEx": "^[^a-zA-Z]{1,10}$",
                  "ValidationString": "Please enter item description in Chinese up to
                                        10 characters",
                },
              ],
              "DataType": "String",
              "InputDisplayText": "Item Description(Chinese)",
              "InputTarget": "ITEM_LEVEL",
              "IsRequired": true,
              "RestrictedSetValues": Array [],
              "StoredValue": Object {
                "DataType": "STRING",
                "ValueAsBoolean": false,
                "ValueAsInteger": 0,
                "ValueAsString": "è¡£æœ�",
              },
            },
          },
          Object {
            "AdditionalInputFieldName": "ITEM_VALUE",
            "SellerInputDefinition": Object {
              "Constraints": Array [
                Object {
                  "ValidationRegEx": "^[0-9]{1,3}(.[0-9]{1,2})?$",
                  "ValidationString": "Please enter item value in USD (max 6
                                        digits)",
                },
              ],
              "DataType": "String",
              "InputDisplayText": "Item Value(USD)",
              "InputTarget": "ITEM_LEVEL",
              "IsRequired": true,
              "RestrictedSetValues": Array [],
              "StoredValue": Object {
                "DataType": "CURRENCY",
                "ValueAsBoolean": false,
                "ValueAsInteger": 0,
                "ValueAsString": "200",
              },
            },
          },
          Object {
            "AdditionalInputFieldName": "HS_CODE",
            "SellerInputDefinition": Object {
              "Constraints": Array [
                Object {
                  "ValidationRegEx": "^[0-9a-zA-Z -]{1,10}?$",
                  "ValidationString": "Please enter HS Code containing letters and
                                        numbers only up to 10 characters",
                },
              ],
              "DataType": "String",
              "InputDisplayText": "HS Code",
              "InputTarget": "ITEM_LEVEL",
              "IsRequired": false,
              "RestrictedSetValues": Array [],
            },
          },
          Object {
            "AdditionalInputFieldName": "ITEM_DESCRIPTION",
            "SellerInputDefinition": Object {
              "Constraints": Array [
                Object {
                  "ValidationRegEx": "^[a-zA-Z0-9-._&amp; ]{3,25}$",
                  "ValidationString": "Please enter item description using
                                        alphanumeric characters (max 25
                                        characters)",
                },
              ],
              "DataType": "String",
              "InputDisplayText": "Item Description",
              "InputTarget": "ITEM_LEVEL",
              "IsRequired": true,
              "RestrictedSetValues": Array [],
              "StoredValue": Object {
                "DataType": "STRING",
                "ValueAsBoolean": false,
                "ValueAsInteger": 0,
                "ValueAsString": "Diapers",
              },
            },
          },
        ],
        "Asin": "B004U9TL66",
      },
    ],
    "ShipmentLevelFields": Array [
      Object {
        "AdditionalInputFieldName": "NON_DELIVERABLE_INSTRUCTIONS",
        "SellerInputDefinition": Object {
          "Constraints": Array [],
          "DataType": "String",
          "InputDisplayText": "Non-Deliverable Instructions",
          "InputTarget": "SHIPMENT_LEVEL",
          "IsRequired": true,
          "RestrictedSetValues": Array [
            "Abandon",
            "Return",
          ],
          "StoredValue": Object {
            "DataType": "STRING",
            "ValueAsBoolean": false,
            "ValueAsInteger": 0,
            "ValueAsString": "ABANDON",
          },
        },
      },
      Object {
        "AdditionalInputFieldName": "SENDER_ADDRESS_TRANSLATED",
        "SellerInputDefinition": Object {
          "Constraints": Array [
            Object {
              "ValidationRegEx": "putjust5digitRegex",
              "ValidationString": "Please enter Ship From Address in
                                Chinese",
            },
          ],
          "DataType": "Address",
          "InputDisplayText": "Ship From Address(Chinese)",
          "InputTarget": "SHIPMENT_LEVEL",
          "IsRequired": true,
          "RestrictedSetValues": Array [],
        },
      },
    ],
  },
  Object {
    "quotaMax": 1000,
    "quotaRemaining": 999,
    "quotaResetOn": 2020-04-06T10:22:23.582Z,
    "requestId": "0",
    "timestamp": 2020-05-06T09:22:23.582Z,
  },
]
`;

exports[`merchant-fulfillment getAdditionalSellerInputs should properly match complete response structure from c# 1`] = `
Array [
  Object {
    "ItemLevelFieldsList": Array [
      Object {
        "AdditionalInputs": Array [
          Object {
            "AdditionalInputFieldName": "String",
            "SellerInputDefinition": Object {
              "Constraints": Array [
                Object {
                  "ValidationRegEx": "String",
                  "ValidationString": "String",
                },
              ],
              "DataType": "String",
              "InputDisplayText": "String",
              "InputTarget": "String",
              "IsRequired": true,
              "RestrictedSetValues": Array [
                "String",
              ],
              "StoredValue": Object {
                "DataType": "String",
                "ValueAsAddress": Object {
                  "AddressLine1": "String",
                  "AddressLine2": "String",
                  "AddressLine3": "String",
                  "City": "String",
                  "CountryCode": "String",
                  "DistrictOrCounty": "String",
                  "Email": "String",
                  "Name": "String",
                  "Phone": "String",
                  "PostalCode": "String",
                  "StateOrProvinceCode": "String",
                },
                "ValueAsBoolean": true,
                "ValueAsCurrency": Object {
                  "CurrencyCode": "String",
                },
                "ValueAsDimension": Object {},
                "ValueAsInteger": 1,
                "ValueAsString": "String",
                "ValueAsWeight": Object {
                  "Unit": "String",
                  "Value": 100,
                },
              },
            },
          },
        ],
        "Asin": "String",
      },
    ],
    "ShipmentLevelFields": Array [
      Object {
        "AdditionalInputFieldName": "String",
        "SellerInputDefinition": Object {
          "Constraints": Array [
            Object {
              "ValidationRegEx": "String",
              "ValidationString": "String",
            },
          ],
          "DataType": "String",
          "InputDisplayText": "String",
          "InputTarget": "String",
          "IsRequired": true,
          "RestrictedSetValues": Array [
            "String",
          ],
          "StoredValue": Object {
            "DataType": "String",
            "ValueAsAddress": Object {
              "AddressLine1": "String",
              "AddressLine2": "String",
              "AddressLine3": "String",
              "City": "String",
              "CountryCode": "String",
              "DistrictOrCounty": "String",
              "Email": "String",
              "Name": "String",
              "Phone": "String",
              "PostalCode": "String",
              "StateOrProvinceCode": "String",
            },
            "ValueAsBoolean": true,
            "ValueAsCurrency": Object {
              "CurrencyCode": "String",
            },
            "ValueAsDimension": Object {},
            "ValueAsInteger": 1,
            "ValueAsString": "String",
            "ValueAsWeight": Object {
              "Unit": "String",
              "Value": 100,
            },
          },
        },
      },
    ],
  },
  Object {
    "quotaMax": 1000,
    "quotaRemaining": 999,
    "quotaResetOn": 2020-04-06T10:22:23.582Z,
    "requestId": "0",
    "timestamp": 2020-05-06T09:22:23.582Z,
  },
]
`;

exports[`merchant-fulfillment getEligibleShippingServices returns lists of shipping services if succesful with additional input 1`] = `
Array [
  Object {
    "RejectedShippingServiceList": Array [
      Object {
        "CarrierName": "CHINA_POST",
        "RejectionReasonCode": "INELIGIBLE",
        "RejectionReasonMessage": "This ship method is not eligible for this
                    order.",
        "ShippingServiceId": "CHINA_POST_E_PACKET",
        "ShippingServiceName": "China Post e-Packet",
      },
    ],
    "ShippingServiceList": Array [
      Object {
        "AvailableLabelFormats": Array [
          "ShippingServiceDefault",
          "PDF",
        ],
        "CarrierName": "CHINA_POST",
        "EarliestEstimatedDeliveryDate": 2019-08-27T18:00:00.000Z,
        "LatestEstimatedDeliveryDate": 2019-08-27T18:00:00.000Z,
        "Rate": Object {
          "CurrencyCode": "CNY",
        },
        "RequiresAdditionalSellerInputs": true,
        "ShipDate": 2019-08-08T07:00:00.000Z,
        "ShippingServiceId": "CHINA_POST_E_EMS",
        "ShippingServiceName": "CHINA_POST_E_EMS",
        "ShippingServiceOfferId": "NH0sNBYSrqUP2+5yJdI14yLsCwFfk2413sYQHKbp0ILv8xkJemzkFK/wgUXeC9y4j6dNKuYn3EpQh7YJSc4gvPUfNDsISoAi9tCikbPjI5KOajdY75KvPLXPEC57Od/gAlOQv07JravlVU6oURFoVtc8nwZDviBPfA43owEPmxPglPTttIuS9LfcI0ky/kIP/OT7h+acfQj8JxAtomwX1WUYGZSbzkSL9c6ic5CAeriif1Kr0k8Y0RuloSZs2ZQwOv//nLvV3ZCZic8vU8cGVGHExM9sMKamBDopoh0Kx2JHcDwWHDcKB+lOntkABC84",
        "ShippingServiceOptions": Object {
          "CarrierWillPickUp": true,
          "DeliveryExperience": "DeliveryConfirmationWithoutSignature",
        },
      },
    ],
    "TemporarilyUnavailableCarrierList": Array [
      Object {
        "CarrierName": "DPD",
      },
    ],
    "TermsAndConditionsNotAcceptedCarrierList": Array [
      Object {
        "CarrierName": "DHL",
      },
    ],
  },
  Object {
    "quotaMax": 1000,
    "quotaRemaining": 999,
    "quotaResetOn": 2020-04-06T10:22:23.582Z,
    "requestId": "0",
    "timestamp": 2020-05-06T09:22:23.582Z,
  },
]
`;

exports[`merchant-fulfillment getEligibleShippingServices returns lists of shipping services if succesful with no additional input 1`] = `
Array [
  Object {
    "RejectedShippingServiceList": Array [
      Object {
        "CarrierName": "UPS",
        "RejectionReasonCode": "INELIGIBLE",
        "RejectionReasonMessage": "This ship method is not eligible for this
                    order.",
        "ShippingServiceId": "UPS_PTP_3DS",
        "ShippingServiceName": "UPS Three Day Saver",
      },
    ],
    "ShippingServiceList": Array [
      Object {
        "AvailableLabelFormats": Array [
          "ZPL203",
          "ShippingServiceDefault",
          "PDF",
          "PNG",
        ],
        "CarrierName": "UPS",
        "EarliestEstimatedDeliveryDate": 2018-10-17T18:00:00.000Z,
        "LatestEstimatedDeliveryDate": 2018-10-17T18:00:00.000Z,
        "Rate": Object {
          "CurrencyCode": "USD",
        },
        "RequiresAdditionalSellerInputs": false,
        "ShipDate": 2018-10-15T23:41:12.000Z,
        "ShippingServiceId": "UPS_PTP_GND",
        "ShippingServiceName": "UPS Ground",
        "ShippingServiceOfferId": "ly51yzA1nAXlJzmYlKaH+7WbKOWz2BTujIOPx5PU8luilmZYK/JFhbx177aKey8MdZcG90uABbT5q8WAgPI+uxcxx/XDtIbI7c161j5spbfsiPTwIJg4Bk66bnP6Ip+JJtuSU5++rAdv4mzXvKpisLGYGrc+FuJvBAjTersJhq4=",
        "ShippingServiceOptions": Object {
          "CarrierWillPickUp": false,
          "DeliveryExperience": "DeliveryConfirmationWithoutSignature",
        },
      },
    ],
    "TemporarilyUnavailableCarrierList": Array [
      Object {
        "CarrierName": "DYNAMEX",
      },
      Object {
        "CarrierName": "USPS",
      },
      Object {
        "CarrierName": "DHLECOMMERCE",
      },
      Object {
        "CarrierName": "SELF_DELIVERY",
      },
      Object {
        "CarrierName": "ONTRAC",
      },
      Object {
        "CarrierName": "DHLMX",
      },
      Object {
        "CarrierName": "DHLM",
      },
      Object {
        "CarrierName": "FEDEX",
      },
    ],
    "TermsAndConditionsNotAcceptedCarrierList": Array [
      Object {
        "CarrierName": "AMZN_US",
      },
      Object {
        "CarrierName": "ONTRAC",
      },
    ],
  },
  Object {
    "quotaMax": 1000,
    "quotaRemaining": 999,
    "quotaResetOn": 2020-04-06T10:22:23.582Z,
    "requestId": "0",
    "timestamp": 2020-05-06T09:22:23.582Z,
  },
]
`;

exports[`merchant-fulfillment getEligibleShippingServices returns lists of shipping services if succesful with test from C# mocks 1`] = `
Array [
  Object {
    "RejectedShippingServiceList": Array [
      Object {
        "CarrierName": "String",
        "RejectionReasonCode": "String",
        "RejectionReasonMessage": "String",
        "ShippingServiceId": "String",
        "ShippingServiceName": "String",
      },
    ],
    "ShippingServiceList": Array [
      Object {
        "AvailableLabelFormats": Array [
          "String",
        ],
        "CarrierName": "String",
        "EarliestEstimatedDeliveryDate": 1969-07-21T02:56:03.000Z,
        "LatestEstimatedDeliveryDate": 1969-07-21T02:56:03.000Z,
        "Rate": Object {
          "CurrencyCode": "String",
        },
        "RequiresAdditionalSellerInputs": true,
        "ShipDate": 1969-07-21T02:56:03.000Z,
        "ShippingServiceId": "String",
        "ShippingServiceName": "String",
        "ShippingServiceOfferId": "String",
        "ShippingServiceOptions": Object {
          "CarrierWillPickUp": true,
          "DeclaredValue": Object {
            "CurrencyCode": "String",
          },
          "DeliveryExperience": "DeliveryConfirmationWithAdultSignature",
          "LabelFormat": "String",
        },
      },
    ],
    "TemporarilyUnavailableCarrierList": Array [
      Object {
        "CarrierName": "String",
      },
    ],
    "TermsAndConditionsNotAcceptedCarrierList": Array [
      Object {
        "CarrierName": "String",
      },
    ],
  },
  Object {
    "quotaMax": 1000,
    "quotaRemaining": 999,
    "quotaResetOn": 2020-04-06T10:22:23.582Z,
    "requestId": "0",
    "timestamp": 2020-05-06T09:22:23.582Z,
  },
]
`;

exports[`merchant-fulfillment getServiceStatus returns a parsed model when the status response is valid 1`] = `
Array [
  Object {
    "Status": "GREEN",
    "Timestamp": "2020-05-06T08:22:23.582Z",
  },
  Object {
    "quotaMax": 1000,
    "quotaRemaining": 999,
    "quotaResetOn": 2020-04-06T10:22:23.582Z,
    "requestId": "0",
    "timestamp": 2020-05-06T09:22:23.582Z,
  },
]
`;
